src/bionics.cpp

```diff
@@ -717,12 +714,25 @@ void player::process_bionic( int b )
     } else if( bio.id == "bio_remote" ) {
         if( g->remoteveh() == nullptr && get_value( "remote_controlling" ) == "" ) {
             bio.powered = false;
-            add_msg( m_warning, _("Your %s has lost connection and is turning off."),
+            add_msg( m_warning, _( "Your %s has lost connection and is turning off." ),
                      bionics[bio.id].name.c_str() );
         }
-    } else if (bio.id == "bio_hydraulics") {
+    } else if( bio.id == "bio_hydraulics" ) {
         // Sound of hissing hydraulic muscle! (not quite as loud as a car horn)
-        sounds::sound( pos(), 19, _("HISISSS!"));
+        sounds::sound( pos(), 19, _( "HISISSS!" ) );
+    } else if( bio.id == "bio_nanobots" ) {
+        for( int i = 0; i < num_hp_parts; i++ ) {
+            if( power_level >= 5 && hp_cur[i] < hp_max[i] ) {
+                heal( ( hp_part )i, 1 );
+                charge_power( -5 );
+            }
+        }
+        for( int i = 0; i < num_bp; i++ ) {
```

Comment:
Why iterate over the body parts if you are going to remove the effect from all at once anyway (calling `remove_effect` without a body part parameter does that).

Also, you can just call `remove_effect` without checking for the existence of the effect. It will return `true` if the effect was removed (implying that it was actually there) and `false` if the effect was not there at all.